1. extraction/gnaf.py

```diff
@@ -0,0 +1,116 @@
+import pandas as pd
```

Comment:
Please use polars instead of pandas

Reply:
You can implement Polars as long as it works, Pandas works.